test/cypress/integration/simplified-account/service-settings/card-payments/index.cy.js

```diff
@@ -73,28 +81,41 @@ describe('Card payments page', () => {
           cy.get('.govuk-summary-list__row').eq(0).within(() => {
             cy.get('.govuk-summary-list__key').should('contain', 'Collect billing address')
             cy.get('.govuk-summary-list__value').should('contain', 'Off')
-            cy.get('.govuk-summary-list__actions').should('contain', 'Change')
+            cy.get('.govuk-summary-list__actions')
+              .find('a')
+              .should('have.attr', 'href', `${pageUrl}/collect-billing-address`)
+              .should('contain', 'Change')
           })
           cy.get('.govuk-summary-list__row').eq(1).within(() => {
             cy.get('.govuk-summary-list__key').should('contain', 'Default billing address country')
             cy.get('.govuk-summary-list__value').should('contain', 'None')
-            cy.get('.govuk-summary-list__actions').should('contain', 'Change')
+            cy.get('.govuk-summary-list__actions')
+              .find('a')
+              .should('have.attr', 'href', `${pageUrl}/default-billing-address-country`)
+              .should('contain.text', 'Change')
```

Comment:
checking the href for correctness is an improvement but I think actually following the link and then navigating back to this page would be a more comprehensive test. There's an example of a test that does something similar here:

https://github.com/alphagov/pay-selfservice/blob/52379b3da97d925141c86582254cdf135f49ba7f/test/cypress/integration/simplified-account/service-settings/stripe-details/task-summary-test-helpers.js#L25

Reply:
Updated with navigation to the details and back.